File: frontend/src/types/exoplanet.ts
export interface ExoplanetData {
  id: string;
  name: string;
  // Core fields that AI should always detect
  discovery_method: 'Transit' | 'Radial Velocity' | 'Direct Imaging' | 'Microlensing' | 'Astrometry';
  discovery_year: number;
  
  // Optional fields that AI might not always detect
  mission?: string;
  radius?: number; // Earth radii
  mass?: number; // Earth masses
  distance?: number; // light years
  orbital_period?: number; // days
  temperature?: number; // Kelvin
  host_star?: string;
  
  // AI confidence scores (optional)
  radius_confidence?: number; // 0-1 confidence score
  mass_confidence?: number; // 0-1 confidence score
  distance_confidence?: number; // 0-1 confidence score
  
  // Data visualization
  graph_data?: Array<{
    time: number;
    brightness?: number;
    velocity?: number;
  }>;
  
  // Additional info
  description?: string;
  discovery_paper?: string;
  
  // AI model metadata
  ai_model_version?: string;
  detection_confidence?: number; // Overall confidence
  last_updated?: string; // When AI model last analyzed this
}

export interface ExoplanetFilters {
  search: string;
  yearRange: [number, number];
  discoveryMethod: string;
  sizeRange: [number, number];
  mission: string;
}

File: exoplanet-api/app/models/exoplanet.py
from pydantic import BaseModel
from typing import List

# --- Exoplanet Models ---

class ExoplanetSummary(BaseModel):
    """A summary model for the list of all exoplanets."""
    id: str
    name: str
    distance: float
    mass: float
    detection_confidence: float
    model_version: str

class GraphDataPoint(BaseModel):
    """A single data point for the transit graph."""
    time: float
    brightness: float

class ExoplanetDetail(BaseModel):
    """The full detailed model for a single exoplanet."""
    id: str
    name: str
    discovery_method: str
    discovery_year: int
    radius: float
    radius_confidence: float
    mass: float
    mass_confidence: float
    distance: float
    distance_confidence: float
    orbital_period: float
    temperature: int
    host_star: str
    description: str
    graph_data: List[GraphDataPoint]
    ai_model_version: str
    detection_confidence: float
    last_updated: str

# --- Mission Models ---

class MissionReport(BaseModel):
    """Model for returning mission analysis data."""
    name: str
    classification_report: str
    roc_curve_url: str